Softeq.XToolkit.Common.Tests/Extensions/EnumerableExtensionsTests/EnumerableExtensionsTests.cs

```diff
@@ -0,0 +1,148 @@
+            IEnumerable<int> enumerable = null;
+
+            Assert.Throws<NullReferenceException>(() =>
+            {
+                enumerable.Apply(null);
+            });
+        }
+
+        [Fact]
+        public void Apply_NullEnumerableNotNullAction_ThrowsNullReferenceException()
+        {
+            IEnumerable<int> enumerable = null;
+
+            Assert.Throws<NullReferenceException>(() =>
+            {
+                enumerable.Apply(_testAction);
+            });
+        }
+
+        [Fact]
+        public void Apply_NotNullEnumerableNullAction_ThrowsNullReferenceException()
+        {
+            Assert.Throws<NullReferenceException>(() =>
+            {
+                _testEnumerable.Apply(null);
+            });
+        }
+
+        [Fact]
+        public void Apply_EmptyEnumerableNotNullAction_DoesNotThrow()
+        {
+            IEnumerable<int> enumerable = new List<int>();
+
+            enumerable.Apply(_testAction);
+
+            _testAction.DidNotReceive().Invoke(Arg.Any<int>());
+        }
+
+        [Fact]
+        public void Apply_NotNullParams_ActionIsCalledForEveryElement()
+        {
+            _testAction.ClearReceivedCalls();
+
+            _testEnumerable.Apply(_testAction);
+
+            _testAction.ReceivedWithAnyArgs(_testEnumerable.Count()).Invoke(Arg.Any<int>());
+            foreach (var item in _testEnumerable)
+            {
+                _testAction.Received().Invoke(item);
+            }
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(0)]
+        [InlineData(1)]
+        public void Chunkify_NullEnumerable_ThrowsArgumentNullException(int size)
+        {
+            IEnumerable<int> enumerable = null;
+
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                enumerable.Chunkify(size).ToList();
+            });
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(0)]
+        public void Chunkify_IncorrectSize_ThrowsOutOfRangeException(int size)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                _testEnumerable.Chunkify(size).ToList();
+            });
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(10)]
+        public void Chunkify_CorrectSize_ReturnsValidChunks(int size)
+        {
+            var expectedTotalCount = (int) Math.Ceiling(_testEnumerable.Count() / (double) size);
+            var lastChunkCount = _testEnumerable.Count() % size;
+
+            var result = _testEnumerable.Chunkify(size).ToList();
+
+            Assert.Equal(expectedTotalCount, result.Count());
+
+            var initialArray = _testEnumerable.ToArray();
+            for (var i = 0; i < result.Count(); i++)
+            {
+                var chunk = result[i];
+                var expectedSize = lastChunkCount != 0 && i == expectedTotalCount - 1 ? lastChunkCount : size;
```

Comment:
be better to avoid duplicate internal details, and compare arrays as-is:

```
input: 12345
output:
12
34
5
```

Reply:
Agree! Will rework this test case